daemon/snap.go

```diff
@@ -338,12 +342,68 @@ func fillComponentInfo(about aboutSnap) []client.Component {
 	return comps
 }
 
-// snapIcon tries to find the icon inside the snap
+// snapIcon tries to find the icon inside the snap.
 func snapIcon(info snap.PlaceInfo) string {
 	found, _ := filepath.Glob(filepath.Join(info.MountDir(), "meta", "gui", "icon.*"))
-	if len(found) == 0 {
+	if len(found) > 0 {
+		return found[0]
```

Comment:
Maybe we should check first if there is an icon between all those that has a name that fits the snap name...

Reply:
Also, if no icon name fits the snap name, I would just "save for later" this list and check the .desktop files...